src/taipy/core/scenario/scenario.py

```diff
@@ -100,7 +100,7 @@ def __init__(
         self._primary_scenario = is_primary
         self._tags = tags or set()
         self._properties = _Properties(self, **properties)
-        self._pipelines: Union[List[Pipeline], List] = pipelines or []
+        self._pipelines: Union[Dict[str, Pipeline], Dict[str, PipelineId], Dict] = pipelines or {}
```

Comment:
```suggestion
        self._pipelines: Union[Dict[str, Pipeline], Dict[str, PipelineId]] = pipelines or {}
```
Again, there is an additional `Dict` type hint. Not sure if it's necessary/

Reply:
mypy is complaining about it 🤣  as the default value is { }

Reply:
Ah ... mypy again

Reply:
if we replace {} by Dict(), is it better?

Reply:
I tried but it's still complaining...